test/ffmpeg-qsv/encode/10bit/vp9.py

```diff
@@ -10,18 +10,21 @@
 
 spec = load_test_spec("vp9", "encode", "10bit")
 
+@slash.requires(*have_ffmpeg_encoder("vp9_qsv"))
+@slash.requires(*have_ffmpeg_decoder("vp9_qsv"))
 class VP9EncoderTest(EncoderTest):
   def before(self):
     vars(self).update(
-      codec   = "vp9",
-      ffencoder   = "vp9_qsv",
-      ffdecoder  ="vp9_qsv",
+      codec     = "vp9",
+      ffencoder = "vp9_qsv",
+      ffdecoder = "vp9_qsv",
     )
     super(VP9EncoderTest, self).before()
 
   def get_file_ext(self):
     return "ivf"
 
+@slash.requires(*platform.have_caps("vdenc", "vp9_10"))
```

Comment:
I plan to consolidate these in a different PR.  Since many codec tests have low-power vs. non-low-power test types I wanted it to stay consistent, even if only one type is currently defined in a codec test.  Both types might get defined in the future.  The plan is to create new base classes in another PR:

```python
@slash.requires(*have_ffmpeg_encoder("vp9_qsv"))
@slash.requires(*have_ffmpeg_decoder("vp9_qsv"))
class VP9EncoderBaseTest (EncoderTest):
  # setup common params and stuff related to this class requirements

@slash.requires(*platform.have_caps("encode", "vp9_10"))
class VP9EncoderTest (VP9EncoderBaseTest):
  # setup non-low-power params and stuff related to this class requirements

@slash.requires(*platform.have_caps("vdenc", "vp9_10"))
class VP9EncoderLPTest (VP9EncoderBaseTest):
  # setup low-power params and stuff related to this class requirements

class cbr_lp(VP9EncoderLPTest):
  # define test method(s) and setup cbr variant params

...
```

Reply:
The exception is jpeg and mpeg2 codecs.  AFAIK, those will always have one type.  But I could be wrong.

Reply:
See my next patch https://github.com/uartie/vaapi-fits/commit/f7d9c37a5125abf78b2b1118e734e962c4ca6d10.  I will submit after this patch merges.

Reply:
I guess it's better if submit those changes along with this PR once I finish them... I will change this to draft until those are ready.

Reply:
all updated.